Comments/components/commentMorph.js

```diff
@@ -26,42 +25,67 @@ export class CommentGroupMorph extends Morph {
     this.ui.groupNameLabel.textString = this.referenceMorph.name;
   }
 
-  async getCommentMorphs (comments) {
-    const commentMorphs = [];
-    await Promise.all(comments.map(async (comment) => {
-      const commentMorph = await resource('part://CommentComponents/comment morph master').read();
-      this.commentIndicators.push(CommentIndicator.for(this.referenceMorph, comment));
-      commentMorph.initialize(comment, this.referenceMorph);
-      commentMorphs.push(commentMorph);
-    }));
-    this.ui.commentCountLabel.textString = commentMorphs.length;
-    return commentMorphs;
+  async addCommentMorph (comment) {
+    const commentMorph = await resource('part://CommentComponents/comment morph master').read();
+    const commentIndicator = CommentIndicator.for(this.referenceMorph, comment);
+    this.commentIndicators.push(commentIndicator);
+    commentMorph.initialize(comment, this.referenceMorph);
+    this.ui.commentMorphContainer.addMorph(commentMorph);
+    this.ui.commentMorphContainer.extent = pt(0, 0);
+    this.updateCommentCountLabel();
   }
 
-  async refreshCommentMorphs () {
-    const commentMorphs = await this.getCommentMorphs(this.referenceMorph.comments);
-    this.ui.commentMorphContainer.submorphs = commentMorphs;
+  onOwnerChanged () {
+    // called when comment groups enter or exit the screen
+    super.onOwnerChanged();
+    if (CommentBrowser.isOpen()) {
+      this.showCommentIndicators();
+    } else {
+      this.hideCommentIndicators();
+    }
+  }
+
+  updateCommentCountLabel () {
+    this.ui.commentCountLabel.textString = this.getCommentMorphCount();
+  }
+
+  getCommentMorphCount () {
+    return this.ui.commentMorphContainer.submorphs.length;
+  }
+
+  async removeCommentMorph (comment) {
+    this.ui.commentMorphContainer.submorphs.forEach((commentMorph) => {
+      if (commentMorph.comment.timestamp == comment.timestamp) {
```

Comment:
This needs to be adjusted to be consistent with the comparisons as @SilvanVerhoeven and I introduced them while refactoring Comments to be a class. Should probably be done after merging, just as a warning :)